src/main/host/descriptor/eventfd.rs

```diff
@@ -212,6 +212,7 @@ impl EventFd {
         self.state
     }
 
+    // SAFETY: This function is not idempotent. Call it only when the counter changes.
     fn update_state(&mut self, cb_queue: &mut CallbackQueue) {
```

Comment:
This seems like it could lead to accidental bugs, so I think it would be nicer if this method took a bool flag, and only flips the `INPUT_BUFFER_PARITY` bit if the flag is true. This would force the caller to consider this behaviour, even if we currently always pass `true`. (And as mentioned in a different comment, the `INPUT_BUFFER_PARITY` doesn't really represent a state anyways, so I think handling it separately is okay.)

Reply:
Sounds like a good idea.